Controller/CommentController.java

```diff
@@ -0,0 +1,118 @@
+package Controller;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+
+import javax.swing.BorderFactory;
+import java.awt.BorderLayout;
+import javax.swing.BoxLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+
+import javax.swing.JLabel;
+
+import java.awt.GridLayout;
+import javax.swing.JPanel;
+import javax.swing.JTable;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import javax.swing.table.DefaultTableModel;
+import javax.swing.table.TableColumnModel;
+import javax.swing.text.AttributeSet.ColorAttribute;
+import javax.xml.stream.events.Comment;
+
+import Model.Administrator;
+import Model.CommentList;
+import Model.CommentModel;
+import Model.Lecturer;
+import Model.Project;
+import Model.ProjectList;
+import Model.Student;
+import Model.User;
+import View.CommentView;
+import View.ProjectView;
+
+public class CommentController {
+    private User user;
+    private Project projectModel;
+    private CommentModel model;
+    private CommentView commentView;
+    private CommentList commentList;
+
+
+
+    public CommentController(CommentModel commentModel, CommentView view,User user ,Project currentProject){
+        this.model = commentModel;
+        this.commentView = view;
```

Comment:
pls try to avoid passing a view into a controller constructor, so that it is cleaner in MainController